File: model/ContaCorrente.cs
using System;

namespace SistemaBancario
{
    public class ContaCorrente: Conta
    {
        public ContaCorrente(int numero, Titular titular) : base(numero, titular)
        {
        }
        public ContaCorrente()
        {
        }
        public override void Depositar(decimal valor)
        {
            if (valor <= 0)
                throw new ArgumentException("O valor do depósito deve ser positivo.");

            saldo += valor * 0.99m; // Taxa de depósito de 1%
        }

        public override void Sacar(decimal valor)
        {
            decimal valorComTaxa = valor * 1.01m; // Taxa de saque de 1%
            if (valor <= 0)
                throw new ArgumentException("O valor do saque deve ser positivo.");
            if (valorComTaxa > saldo)
                throw new InvalidOperationException("Saldo insuficiente para o saque.");

            saldo -= valorComTaxa;
        }

        public override string ToString()
        {
            return $"Conta Corrente - {base.ToString()}";
        }
    }
}

File: services/ContaCorrente.php
<?php
require_once 'Conta.php';
class ContaCorrente extends Conta
{
    public function depositar(float $valor): void
    {
        if ($valor <= 0) {
            throw new Exception("O valor do depósito deve ser positivo.");
        }

       
        $valorComTaxa = $valor * 0.99;
        $this->saldo += $valorComTaxa;
    }

    public function sacar(float $valor): void
    {
        if ($valor <= 0) {
            throw new Exception("O valor do saque deve ser positivo.");
        }

        $valorComTaxa = $valor * 1.01;
        
        if ($valorComTaxa > $this->saldo) {
            throw new Exception("Saldo insuficiente para o saque.");
        }

        $this->saldo -= $valorComTaxa;
    }

    public function __toString(): string
    {
        return "Conta Corrente - " . parent::__toString();
    }
}

?>